frontend/sac-mobile/app/(auth)/_components/tag-form.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+import { useForm } from 'react-hook-form';
+import { Alert, SafeAreaView, ScrollView, Text, View } from 'react-native';
+
+import { router } from 'expo-router';
+
+import { ZodError } from 'zod';
+
+import { Button } from '@/components/button';
+import Error from '@/components/error';
+import { categories } from '@/lib/const';
+import { Category } from '@/types/categories';
+
+type TagsData = {
+    tags: String[];
+};
+
+// combine all tags of different categories into one for All tab
+let allTags: string[] = [];
```

Comment:
should be in a function and possibly a utility if reused or if it's functionality that can be generalized should be generalized